syntax = "proto3";
package user.auth;

message getUserTokenReq {
  int32 userID = 1;
  string secret = 2;
}

message getUserTokenResp {
  string token = 1;
  int64 expireTimeSeconds = 2;
}

message parseTokenReq {
  string token = 1;
}

message parseTokenResp {
  int32 userID = 1;
}

message registerUserReq {
  int32 userID = 1;
  string nickName = 2;
  string faceURL = 3;
}

message registerUserResp {
  bool result = 1;
}

service Auth {
  // 获取用户 token
  rpc getUserToken(getUserTokenReq) returns (getUserTokenResp);
  // 解析 token
  rpc parseToken(parseTokenReq) returns (parseTokenResp);
  // 注册用户
  rpc registerUer(registerUserReq) returns (registerUserResp);
}
